src/category_theory/lifting_properties.lean

```diff
@@ -0,0 +1,208 @@
+    rw ←category.assoc at this,
+    tidy,
+  end
+}
+
+/- The condition of having the rlp with respect to a morphism i is stable under composition-/
+lemma has_right_lifting_property_comp {i : arrow C} {f : X ⟶ Y} {g : Y ⟶ Z}
+(hf : has_lifting_property i (arrow.mk f))
+(hg : has_lifting_property i (arrow.mk g)) : 
+has_lifting_property i (arrow.mk (f ≫ g)) :=
+begin
+  intro sq0, -- a square between i and f ≫ g
+  let sq1 := square_to_second sq0, -- transform this into a square between i and g
+  
+  -- lift i vs. g
+  haveI := hg sq1,
+  have lift_structure_sq1 := arrow.has_lift.struct sq1,
+  have sq1_lift := arrow.lift sq1,
+
+  -- form a square from i to f, using the previously constructed lift 
+  have h3 : sq0.left ≫ (arrow.mk f).hom = i.hom ≫ lift_structure_sq1.lift :=
+  begin
+    rw lift_structure_sq1.fac_left,
+    refl,
+  end,
+
+  -- construct a square i ⟶ f
+  let sq2 : i ⟶ (arrow.mk f) :=
+  {
+    left := sq0.left,
+    right := lift_structure_sq1.lift,
+    w' := begin
+      simp,
+      tidy,
+    end,
+  },
+  
+  -- construct a lift i vs. f
+  haveI := hf sq2,
+  have lift_structure_sq2 := arrow.has_lift.struct sq2,
+  have sq2_lift := arrow.lift sq2,
+  
+  -- show that this lift is a lift of i vs. g ∘ f
+  fconstructor,
+  fconstructor,
+  fconstructor,
+  { simp, exact lift_structure_sq2.lift, },
+  { simp,  },
+  { simp, tidy,
+    rw ←category.assoc,
+    let d := lift_structure_sq2.fac_right,
+    let d' := lift_structure_sq1.fac_right,
+    have : sq0.right = sq1.right := begin tidy, end,
+    rw this,
+    rw ←d',
+    tidy,
+    rw ←category.assoc,
+    rw d,
+  },
+end
+
+/- Right lifting conditions relative to a diagram of arrows in C. -/
+variable {F : D ⥤ arrow C}
+
+def right_lifting_property_rel (p : X ⟶ Y) : Prop := 
+∀ i : D, has_lifting_property ((F.obj) i) (arrow.mk p)
+
+lemma id_has_right_lifting_property' (X : C) :  
+∀ i : D, has_lifting_property ((F.obj) i) (arrow.mk (𝟙 X)) :=
+begin
+  intro i,
+  exact id_has_right_lifting_property ((F.obj) i),
+end
+
+lemma has_right_lifting_property_comp' 
+{f : X ⟶ Y} (hf : ∀ i : D, has_lifting_property ((F.obj) i) (arrow.mk f))
+{g : Y ⟶ Z} (hg : ∀ i : D, has_lifting_property ((F.obj) i) (arrow.mk g)) :
+∀ i : D,  has_lifting_property ((F.obj) i) (arrow.mk (f ≫ g)) :=
+begin
+  intro i,
+  exact has_right_lifting_property_comp (hf i) (hg i),
+end
+
+/- Given a diagram of arrows in C, F : D ⥤ arrow C, we construct the (non-full) subcategory of C 
+spanned by those morphisms that have the right lifting property relative to all maps
+of the form F(i), where i is any object in D. Its objects are the same as the one of C. -/
+def right_lifting_subcat (F : D ⥤ arrow C) := C
+def right_lifting_subcat.X (x : right_lifting_subcat F) : C := x
+
+instance : category (right_lifting_subcat F) :=
+{ hom := λ X Y, { p : X.X ⟶ Y.X // 
+    ∀ {i : D}, has_lifting_property ((F.obj) i) (arrow.mk p) },
+  id := λ X, ⟨ 𝟙 X, id_has_right_lifting_property' X, ⟩,
```

Comment:
Very minor, but usually the style is no spaces on the "insides" of `⟨` and `⟩`.

Reply:
OK.